tools/windows-node-installer/pkg/cloudprovider/aws/ec2_test.go

```diff
@@ -149,11 +142,43 @@ func TestGetMissingRules(t *testing.T) {
 				getOtherPortRules(WINRM_PORT, myIP),
 				getOtherPortRules(sshPort, myIP),
 				getOtherPortRules(rdpPort, myIP),
-			}},
+			},
+		},
+	}
+
+	for _, tt := range testIO {
+		actual := awsProvider.getRulesForSgUpdate(myIP, tt.in, vpcCidr)
+		assert.ElementsMatch(t, tt.expected, actual, tt.name, "awsProvider.getRulesForSgUpdate(), actual values do not match expected")
+	}
+}
+
+// TestExamineRulesInSg tests the examineRulesInSg function
+func TestExamineRulesInSg(t *testing.T) {
```

Comment:
Do you also want to exercise the code path for `hasClusterCIDRRule` instead of not checking it all?

Reply:
+1 on this

Reply:
However, this PR is about fixing the bug and this is out of the scope of doing that.

Reply:
I am going to remove the test case from this PR now so it gets merged with a todo and add a new story for more test coverage.